src/sentry/eventstream/kafka/consumer.py

```diff
@@ -274,6 +274,10 @@ def assignment_callback(consumer, assignment):
                                  for topic, partition in assignment.keys()])
 
         def revocation_callback(consumer, assignment):
+            for item in assignment:
+                # TODO: This should probably also be removed from the state manager.
```

Comment:
This won't lead to any bad behavior on assignment, since the local offset will be set again when the consumer is reassigned the partition: https://github.com/getsentry/sentry/blob/71a4ca9b142cb784fe6713d718a77d540e35a049/src/sentry/eventstream/kafka/consumer.py#L269

It probably would make sense to modify this so that state transitions back to "UNKNOWN" are acceptable (right now these are not supported) when a partition is revoked.